magicctapipe/scripts/lst1_magic/semi_automatic_scripts/job_accounting.py

```diff
@@ -124,30 +127,45 @@ def main():
                 returns = fp.readlines()
                 this_return = len(returns)
                 for line in returns:
-                    file_in, slurm_id, task_id, rc = line.split()
+                    line = line.split()
+                    file_in = line[0]
+                    slurm_id = f"{line[1]}_{line[2]}" if len(line) == 4 else line[1]
+                    rc = line[-1]
                     if rc == "0":
                         this_good += 1
                         # now check accounting
                         if not args.no_accounting:
                             out = run_shell(
-                                f'sacct --format="JobID,CPUTime,MaxRSS" --units=M  -j {slurm_id}_{task_id}| tail -1'
-                            )
-                            _, cpu, mem = out.split()
+                                f'sacct --format="JobID,CPUTime,MaxRSS" --units=M  -j {slurm_id}| tail -1'
+                            ).split()
+                            if len(out) == 3:
+                                _, cpu, mem = out
+                            else:
```

Comment:
should not happen in theory, but better to put `elif len(out) == 4` (I guess it is the other possible option right?) and then an else to e.g. catch that there is an error with the formatting

Reply:
in principle it should be 3 since I ask for them in the format: ```--format="JobID,CPUTime,MaxRSS"```, but I saw that MaxRSS is not always filled in (in particular immediately after the job was finished I was getting empty field there), so I also implemented the case for 2 fields. I do not unpact it then, so if e.g. 4 fields would appear it would also extract CPU as the second one, but since 4 is really unexpected it is more probable that sacct makes something strange and e.g. does not provide any output, so now I added this else statement as you suggested